Conventional/Conventions/Cecil/MustNotResolveCurrentTimeViaDateTimeDirectlyConventionSpecification.cs

```diff
@@ -1,38 +1,46 @@
-﻿using System;
-using System.Linq;
-using Conventional.Conventions;
-using Mono.Cecil;
-using Mono.Cecil.Cil;
-
-namespace Conventional.Cecil.Conventions
-{
-    public class MustNotUseDateTimeNowConventionSpecification : ConventionSpecification
-    {
-        protected override string FailureMessage
-        {
-            get { return "Must not use DateTime.Now. It is called {0} times in this type."; }
-        }
-
-        public override ConventionResult IsSatisfiedBy(Type type)
-        {
-            var nowAssignments =
-                type.ToTypeDefinition()
-                    .Methods
-                    .Where(method => method.HasBody)
-                    .SelectMany(method => method.Body.Instructions)
-                    .Where(
-                        x =>
-                            x.OpCode == OpCodes.Call && x.Operand is MethodReference &&
-                            ((MethodReference) x.Operand).DeclaringType.FullName == "System.DateTime" &&
-                            ((MethodReference) x.Operand).Name == "get_Now")
-                    .ToArray();
-
-            if (nowAssignments.Any())
-            {
-                return ConventionResult.NotSatisfied(type.FullName, FailureMessage.FormatWith(nowAssignments.Count()));
-            }
-
-            return ConventionResult.Satisfied(type.FullName);
-        }
-    }
+﻿using System;
+using System.Linq;
+using Conventional.Conventions;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+
+namespace Conventional.Cecil.Conventions
+{
+    public class MustNotResolveCurrentTimeViaDateTimeDirectlyConventionSpecification : ConventionSpecification
+    {
+        protected override string FailureMessage
+        {
```

Comment:
Perhaps a change to "Must not use DateTime directly. It is used {0} times in {1}". I do like having the aggregate count of violations in this one, I think it would aid in tracking down offenders. I like your suggestion of putting the type name in though, so {1} could be the type name.